docs/application_sources.md

```diff
@@ -101,3 +102,33 @@ value, in the values.yaml such that the value is stable between each comparison.
 ```
 argocd app set redis -p password=abc123
 ```
+
+## Config Management Plugins
+
+Argo CD allows integrating more config management tools using config management plugins. Following changes are required to configure new plugin:
+
+* Make sure required binaries are available in `argocd-repo-server` pod. The binaries can be added via volume mounts or using custom image (see [custom_tools](custom_tools.md)).
+* Register new plugin in `argocd-cm` ConfigMap:
+
+```yaml
+data:
+  configManagementPlugins: |
+    - name: pluginName
+      init:                          # Optional command to initialize application source directory
+        command: ["sample command"]
+        args: ["sample args"]
+      generate:                      # Command to generate manifests YAML
+        command: ["sample command"]
+        args: ["sample args"]
+```
+
+Both `init` and `generate` commands are executed inside application source directory. Commands have access to system environment variables and following additional variables:
```

Comment:
"inside the application source directory"

Reply:
"Commands have access to system environment variables, including:

Reply:
I meant to say here: all system environment variables of `argocd-repo-server` pod + additional variables. `ARGOCD_APP_NAME` and `ARGOCD_APP_NAMESPACE`.